delivery/delivery.go

```diff
@@ -61,13 +61,23 @@ func (h *HTTPHandler) UpdateBook(c *gin.Context) {
 		return
 	}
 
+	// Check if the book exists
+	_, err := h.BookUseCase.GetBookById(id)
+	if err != nil {
+		c.IndentedJSON(http.StatusNotFound, gin.H{"message": "Book not found"})
```

Comment:
Can you explain about `c.IndentedJSON` ?

Reply:
the `c.IndentedJSON` is a gin function to make json output to be indented so its more readable

Reply:
what's the difference between c.JSON and c.IndentedJSON?

Reply:
they are similliar the only difference are `c.JSON` produce an output where the json are not indented while `c.IndentedJSON` produce json that is indented

Reply:
Is there an impact on size and performance?

Reply:
Yes there is an impact because the indentedJson produce additional space but i think the impact is not that significant because it is so small